### PR TITLE
[C4-3233] Gifts Home Page 

### PR DESCRIPTION
**Pull request by char1e5**
### Overview

1. Home page layout
2. Header and footer
3. Button and Logo components
4. Add lint and types

### Ticket

https://opentable.atlassian.net/browse/C4-3233

<details>
<summary>Additional</summary>

#### A/B Experiment

- Prod: http://abtest.prod-uswest2.otenv.com/
- PP: http://abtest.pp-uswest2.otenv.com/

#### Feature Flags

- Prod: https://hobknob.otenv.com/
- PP: https://hobknob-qa.otenv.com/
</details>

<details>
<summary>Screenshots</summary>

#### Desktop:
screen shot 2018-07-13 at 1 09 08 am

#### Mobile:
screen shot 2018-07-13 at 1 09 25 am

</details>

### CODE REVIEW

File: src/app/pages/Layout/styles.css [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  font-family: font-family;
+  color: ash-dark;
+  background-color: white;
+  -webkit-font-smoothing: antialiased;
+  -moz-osx-font-smoothing: grayscale;
+}
+
+.masterContainer {
+  background-color: white;
+  min-width: 20rem;
```
Comment (diffalot) [inline src/app/pages/Layout/styles.css]: this really should be a design token, and in looking over design-tokens, they recommend using [pxtorem](https://github.com/opentable/design-tokens/#pixel-to-rem-conversion)

so, in a later pr, we should replace this with a token, and add the pixel to rem conversion.

File: src/app/components/Footer/index.jsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+
+import styles from './styles.css';
+
+export default function Footer({
```
Comment (matthemsteger) [inline src/app/components/Footer/index.jsx]: JSDoc props for TypeScript and readability

File: src/app/pages/Layout/index.jsx [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  return (
+    <I18nConsumer>
+      {/**
+       * @param {object} props
+       * @param {function(string): string} props.translate
+       */
+      ({ translate }) => (
+        <div className={styles.masterContainer}>
+          <Helmet>
+            <meta httpEquiv='X-UA-Compatible' content='IE=9; IE=8; IE=7; IE=EDGE' />
```
Comment (matthemsteger) [inline src/app/pages/Layout/index.jsx]: I think we only need edge (highest IE mode available) here, not the rest.

File: src/app/pages/Layout/index.jsx [hunk 4]
```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+import Helmet from 'react-helmet';
+// @ts-ignore @todo replace when types are available
+import { I18nConsumer } from '@c4-components/i18n';
+
+import styles from './styles.css';
+import Header from './../../components/Header';
+import Footer from './../../components/Footer';
+
+export default function Layout(props) {
```
Comment (matthemsteger) [inline src/app/pages/Layout/index.jsx]: JSDoc

File: src/app/pages/Layout/styles.css (hunk 1 above)
Comment (char1e5) [reply src/app/pages/Layout/styles.css]: Added @diffalot

**Pull request merged**